src/mainwindow.cpp

```diff
@@ -75,12 +76,44 @@ MainWindow::~MainWindow()
 }
 
 void MainWindow::toggleFullScreen() {
-    if (isFullScreen())
+    if (isFullScreen()) {
+        QApplication::instance()->removeEventFilter(this);
+        showTabAndTop();
         showNormal();
-    else
+    }
+    else {
+        QApplication::instance()->installEventFilter(this);
+        hideTabAndTop();
         showFullScreen();
+    }
+}
+
+void MainWindow::hideTabAndTop() {
+    getTabBar()->hide();
+    getTopWidget()->hide();
+}
+
+void MainWindow::showTabAndTop() {
+    getTabBar()->show();
+    getTopWidget()->show();
+}
+
+bool MainWindow::eventFilter(QObject* /*object*/, QEvent* event)
+{
+    if (event->type() == QEvent::MouseMove && isFullScreen())
+    {
+        const auto mouseEvent = static_cast<QMouseEvent*>(event);
+        if (mouseEvent->y() == 0) {
+            showTabAndTop();
+        } else if(mouseEvent->y() >= 150) {
```

Comment:
How did you select this value? In my environment it is noticeably below the bottom of the tab bar.

Reply:
The noticeably below was intentional - that's how Firefox does it also. Though value selection was not optimal, found another way this time.